API_Contracts.md

```diff
@@ -72,6 +74,28 @@ Code 200
 
 ---
 
+Used for Login
+
+**`POST /api/v1/users/tokens`**
+
+Body
+```json
+{
+  "email": "string",
+  "username": "string",
```

Comment:
If we are using email or username for login, expect it as a single field that may be called an identifier for example. The backend then should decide if its an email or a username.

Reply:
Perhaps we could do such validation at webapp side? Then once input is validated it would be packed in json accordingly. This would create less overhead as API wouldn't need to query database multiple times.

Reply:
We could just add a constraint that usernames may not contain the '@' symbol and then build the database query accordingly.

Reply:
Validation on the backend is necessary while validation on the frontend is just a good addition. Imagine if there was additional frontend, for example, a mobile app. There's no need to query multiple times, it could be simple if check whether its email or username or just chuck it in SQL query something like "... where email=identifier or username=identifier".